rust/src/common/config.rs

```diff
@@ -145,6 +152,40 @@ impl Configuration {
             .collect()
     }
 
+    /// Get the replace datanode on failure policy from configuration
+    pub fn get_replace_datanode_on_failure_policy(
+        &self,
+    ) -> crate::hdfs::replace_datanode::ReplaceDatanodeOnFailure {
+        use crate::hdfs::replace_datanode::{Policy, ReplaceDatanodeOnFailure};
+
+        let enabled = self.get_boolean(
+            DFS_CLIENT_WRITE_REPLACE_DATANODE_ON_FAILURE_ENABLE_KEY,
+            false,
+        );
+        if !enabled {
+            return ReplaceDatanodeOnFailure::new(Policy::Disable, true);
+        }
+
+        let best_effort = self.get_boolean(
+            DFS_CLIENT_WRITE_REPLACE_DATANODE_ON_FAILURE_BEST_EFFORT_KEY,
+            true,
```

Comment:
Should we use the same defaults as HDFS and enable datanode replacement by default without best effort?

Reply:
makes sense.